### ISSUE TITLE
pytest.mark.dependency depends not honored

### ISSUE THREAD

**Issue by jtmoon79** [RKrahl/pytest-dependency#30]
## Using `@pytest.mark.dependency` does not honor the `depends` parameter.

**tl;dr** The dependency check does not work.

Following the example from [readthedocs.io page *usage* section *naming-tests*](https://pytest-dependency.readthedocs.io/en/latest/usage.html#naming-tests), my file looks like

```Python
import pytest

class TestClassNamed(object):

    @pytest.mark.dependency(name="a")
    @pytest.mark.xfail(reason="deliberate fail")
    def test_a(self):
        assert False

    @pytest.mark.dependency(name="b", depends=["a"])
    def test_b(self):
        pass
```

Running this

```
PS > pytest .\test\ -v
===================== test session starts =====================
platform win32 -- Python 3.7.1, pytest-4.2.0, py-1.7.0, pluggy-0.8.1 -- c:\users\user1\.virtualenvs\project1-3kr5jz-l\scripts\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\user1\Downloads\project1\test, inifile:

test/test_1.py::TestClassNamed::test_a XFAIL               [ 50%]
test/test_1.py::TestClassNamed::test_b PASSED              [100%]
```

I expected `TestClassNamed::test_b` to be `SKIPPED` because it `depends` on test named "`a`".  That test `XFAIL`.  However, `TestClassNamed::test_b` was run and `PASSED`.  That was unexpected.

I tried this without the `@pytest.mark.xfail(reason="deliberate fail")` on `test_a`.  However, `TestClassNamed::test_b` still ran and `PASSED`.  That was also unexpected.

### `pip list`

```
PS > pip list
Package        Version
-------------- -------
atomicwrites   1.3.0
attrs          18.2.0
colorama       0.4.1
more-itertools 5.0.0
musicbrainzngs 0.6
mutagen        1.42.0
pip            19.0.1
pluggy         0.8.1
py             1.7.0
pytest         4.2.0
setuptools     40.7.2
six            1.12.0
wheel          0.32.3
```

### pytest version

From the example output

    platform win32 -- Python 3.7.1, pytest-4.2.0, py-1.7.0, pluggy-0.8.1

----

Originally and mistakenly reported in [pytest #4737](https://github.com/pytest-dev/pytest/issues/4737#issue-407490679) (which has been closed).

**Comment by RKrahl** [comment]
I cannot reproduce your problem. I get:
```
$ python3 -m pytest -v              
============================= test session starts ==============================
platform linux -- Python 3.6.5, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python3
cachedir: .cache
rootdir: /usr/src/packages, inifile:
plugins: dependency-0.3.2
collected 2 items                                                              

test_namedclass.py::TestClassNamed::test_a xfail                         [ 50%]
test_namedclass.py::TestClassNamed::test_b SKIPPED                       [100%]

===================== 1 skipped, 1 xfailed in 0.08 seconds =====================
```

From your `pip list` output, I'm missing `pytest-dependency`. Did you install pytest-dependency?

**Comment by jtmoon79** [comment]
**_Doh!_**

I added the decorators `@pytest.mark.dependency(...)` and then `pytest` didn't crash or complain about the decorator .  So I guessed (mistakenly) that meant `pytest-dependency` was included within `pytest`.

I did `pipenv install pytest-dependency --dev` and now it works.

Pardon the noise.

**Comment by RKrahl** [comment]
Yes, those markers in pytest somewhat look like function calls. So one might expect something like a `NameError` if using a marker that has not been defined. But in fact, they are more like arbitrary labels that may be attached to a test. That is why the following will work with no complaint:
```python
import pytest

@pytest.mark.funny_marker_that_has_no_meaning_at_all(sense='absent')
def test_funny():
    pass
```

**Comment by Hamza5** [comment]
> Yes, those markers in pytest somewhat look like function calls. So one might expect something like a `NameError` if using a marker that has not been defined. But in fact, they are more like arbitrary labels that may be attached to a test. That is why the following will work with no complaint:
> 
> ```python
> import pytest
> 
> @pytest.mark.funny_marker_that_has_no_meaning_at_all(sense='absent')
> def test_funny():
>     pass
> ```

I had the same problem because I misspelled the word `dependency` as `dependancy` and the tests were not skipped because of that. Thanks for the clarification.
